twitter/fetch.py

```diff
@@ -0,0 +1,96 @@
+import csv
+import itertools
+from typing import List
+
+import tweepy
+
+from .api import get_default_api
+from .constants import MBTI_TYPES
+from .models import User
+from .log import get_logger
+
+logger = get_logger(__name__)
+
+
+def build_status(status: tweepy.Status) -> dict:
+    return dict(
+        id=status.id,
+        created_at=str(status.created_at),
+        name=status.user.name,
+        screen_name=status.user.screen_name,
+        text=_get_text(status),
+    )
+
+
+def _get_text(status: tweepy.Status) -> dict:
+    text = None
+    truncated = status.truncated
+    error = False
+    if not truncated:
+        text = status.text
+    else:
+        try:
+            text = status._json["extended_tweet"]["full_text"]
+        except KeyError:
+            text = status.text
+            error = True
+    return {"value": text, "truncated": truncated, "error": error}
+
+
+def fetch(
+    screen_name: str, count: int = 50, api: tweepy.API = None
+) -> List[dict]:
+    """Fetch latest tweets of a user.
+
+    Parameters
+    ----------
+    screen_name : str
+        The screen name of the user.
+    count : int, optional
+        The maximum number of tweets to fetch.
+    api : tweepy.API, optional
+    """
+    if api is None:
+        api = get_default_api()
+
+    logger.debug(
+        {"op": "user_timeline", "screen_name": screen_name, "count": count}
+    )
+
+    for status in api.user_timeline(screen_name, count=count):
```

Comment:
`tweepy.error.TweepError: Failed to send request: Only unicode objects are escapable. Got None of type <class 'NoneType'>.` j'ai cette erreur après avoir enlevé la dernière ligne.

Reply:
La première ligne tu veux dire ?

Reply:
oui première

Reply:
Est-ce que tu as suivi le setup des credentials comme indiqué ici: [Retrieving tweets](https://github.com/literate-waffle/maple#retrieving-tweets) ?

Reply:
je me suis créé un compte, maintenant je dois créer une app dans le compte ? ils me demandent un site internet alors je sais pas trop quoi mettre...

Reply:
oui tu dois créer une app pour avoir des credentials, mets ce que tu veux pour le site internet, avec Valentin et Hélène on avait mis mon site perso centrale marseille ;P après ils vont te poser des questions sur le projet pour lequel t'as besoin de credentials

Reply:
ok ça marche maintenant, merci !